tools/ldb_cmd.cc

```diff
@@ -1139,6 +1144,151 @@ void ManifestDumpCommand::DoCommand() {
   }
 }
 
+// ----------------------------------------------------------------------------
+namespace {
+
+void GetAllFileCheckSumInfoFromManifest(Options options, std::string file,
+                                        FileChecksumList* checksum_list) {
+  EnvOptions sopt;
+  std::string dbname("dummy");
+  std::shared_ptr<Cache> tc(NewLRUCache(options.max_open_files - 10,
+                                        options.table_cache_numshardbits));
+  // Notice we are using the default options not through SanitizeOptions(),
+  // if VersionSet::GetAllFileCheckSumInfo depends on any option done by
+  // SanitizeOptions(), we need to initialize it manually.
+  options.db_paths.emplace_back("dummy", 0);
+  options.num_levels = 64;
+  WriteController wc(options.delayed_write_rate);
+  WriteBufferManager wb(options.db_write_buffer_size);
+  ImmutableDBOptions immutable_db_options(options);
+  VersionSet versions(dbname, &immutable_db_options, sopt, tc.get(), &wb, &wc,
```

Comment:
I think you need to call `versions.Recover()`.

Reply:
@riversand963 In this case, the tool just dumps out the existing information from Manifest file. Why versions.Recover() is needed? In this case, we do not have information about the CF and db_id.

Reply:
Oh, I see, `GetFllFileChecksumInfo` scans the MANIFEST, so no need to call Recover.

Reply:
Please see my most recent comment regarding `VersionSet::GetAllFileChecksumInfo`.